internal/db/schema/migrations/oss/postgres/88/01_remove_session_connection_state.up.sql

```diff
@@ -0,0 +1,222 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+
+  -- Remove the session_connection_state table and any related triggers
+  drop trigger update_connection_state_on_closed_reason on session_connection;
+  drop function update_connection_state_on_closed_reason();
+
+  drop trigger insert_session_connection_state on session_connection_state;
+  drop function insert_session_connection_state();
+
+  drop trigger update_session_state_on_termination_reason on session;
+  drop function update_session_state_on_termination_reason();
+
+  drop trigger insert_new_connection_state on session_connection;
+  drop function insert_new_connection_state();
+
+  drop trigger immutable_columns on session_connection_state;
+
+  drop trigger wh_insert_session_connection_state on session_connection_state;
+  drop function wh_insert_session_connection_state();
+
+  drop trigger wh_insert_session_connection on session_connection;
+  drop function wh_insert_session_connection();
+
+  --  If the connected_time_range is null, it means the connection is authorized but not connected.
+  --  If the upper value of connected_time_range is > now() (upper range is infinity) then the state is connected.
+  --  If the upper value of connected_time_range is <= now() then the connection is closed.
+  alter table session_connection
+    add column connected_time_range tstzrange;
+
+  -- Migrate existing data from session_connection_state to session_connection
+   update session_connection
+      set connected_time_range =
+  (select tstzrange(min(start_time), max(start_time))
+     from session_connection_state
+    where session_connection_state.connection_id = session_connection.public_id
+ group by connection_id );
```

Comment:
This is another sub-query that I missed for formating:

```suggestion
  update session_connection
     set connected_time_range = (select tstzrange(min(start_time), max(start_time))
                                   from session_connection_state
                                  where session_connection_state.connection_id = session_connection.public_id
                               group by connection_id);
```